processor/src/test/java/org/mapstruct/ap/test/nestedsourceproperties/ArtistToChartEntryUpdateReverse.java

```diff
@@ -0,0 +1,54 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.nestedsourceproperties;
+
+import org.mapstruct.InheritInverseConfiguration;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.Mappings;
+import org.mapstruct.ap.test.nestedsourceproperties._target.ChartEntry;
+import org.mapstruct.ap.test.nestedsourceproperties.source.Song;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Sjaak Derksen
+ */
+@Mapper
+public abstract class ArtistToChartEntryUpdateReverse {
+
+    public static final ArtistToChartEntryUpdateReverse MAPPER =
+        Mappers.getMapper( ArtistToChartEntryUpdateReverse.class );
+
+    @Mappings({
+
+        @Mapping(target = "songTitle", source = "title"),
+        @Mapping(target = "artistName", source = "artist.name"),
+        @Mapping(target = "recordedAt", source = "artist.label.studio.name"),
+        @Mapping(target = "city", source = "artist.label.studio.city"),
+        @Mapping(target = "position", ignore = true),
+        @Mapping(target = "chartName", ignore = true )
+    })
+    abstract ChartEntry mapForward(Song song);
```

Comment:
> What would happen if there is another property of Artist, say "artist.nickname" which is not covered by this method. How would the inverse mapping method deal with that?

Not. There's no (property) name based matching on nested properties. So as soon as you specify something with `artist` (`artist.name`, `artist.nickname`)`, Mapstruct is happy. The target property is mapped. So, there is no ‘completeness check’.

Do you guys think there’s such a need for such completeness check? It might lead to specifying a whole lot of ‘ignores’ on nesting levels...